controllers:
  gimbal_controller:
    type: rm_gimbal_controllers/Controller
    publish_rate: 100
    yaw:
      joint: "yaw_joint"
      pid: { p: 30.0, i: 0., d: 1.2, i_clamp_max: 0.3, i_clamp_min: -0.3, antiwindup: true, publish_state: true }
    pitch:
      joint: "pitch_joint"
      pid: { p: 20.0, i: 1, d: 0.45, i_clamp_max: 0.4, i_clamp_min: -0.4, antiwindup: true, publish_state: true }
    feedforward:
      gravity: 0.0
      enable_gravity_compensation: false
      mass_origin: [ 0.0,0.0,0.0 ]
    bullet_solver:
      resistance_coff_qd_10: 0.45
      resistance_coff_qd_15: 1.0
      resistance_coff_qd_16: 0.7
      resistance_coff_qd_18: 0.55
      resistance_coff_qd_30: 3.0
      g: 9.81
      delay: 0.1
      dt: 0.001
      timeout: 0.001
      publish_rate: 50
